### ISSUE TITLE
Don't cancel auto-sort when error messages are displayed

### ISSUE THREAD

**Issue by leontristain** [loot/loot#1259]
Hi,

I am currently trying to develop a fully automated workflow for building a modded skyrim setup, and also using it to build my modded skyrim game. I'm quite far along and a lot of pieces are functional. That said, I'm currently running into a potentially blocking issue.

At the end of my automated build process, I automatically run loot with `--auto-sort`. It's been working great so far. However, just now it is failing to complete because "Auto-sort has been cancelled as there is at least one error message displayed."

The error is an incompatibility between `Solitude Skyway SE.esp` and `LegacyoftheDragonborn.esm` in the masterlist. There is an official Skyway patch for this and the masterlist may not have caught up yet, and I also have that Skyway patch in my load order. Though, whether or not I have the correct patch is beside the point. The bigger issue is that by its nature, the LOOT masterlist cannot be 100% correct on these matters (who knows, I might have a personal patch?), which means there ought to be a way to disable this "error on incompatibilities" functionality for when the user knows better.

I understand that some errors, such as cyclic dependencies, means the order is not sortable, so I am not looking for a way to skip all errors, only errors that can be skipped and still lead to sorting. With the incompatibility error, I can proceed manually, so I think I should be able to proceed automatically also.

After searching around in the docs, I couldn't find anything, whether it be config settings, command line flags, that could disable this "error on incompatibilities" functionality, or otherwise help get me unstuck.

I also looked into whether incompatibility can be reverted by a userlist entry, which it does not look possible, since the `inc` field in the yaml is additive. Though this strategy, even if possible, is not as ideal since I'm still at the mercy of masterlist updates breaking automation even if this particular case is resolved.

I really, really, really hope the answer is not going to be "just push buttons manually".

I suppose in the worst case scenario, I can maintain my own masterlist repositories and point LOOT at those, or tell LOOT to not have a masterlist and instead manually pull the masterlist file to set as userlist with overrides that can be specified from my system. Though it would be unfortunate if the solution is so involved for such a simple thing.

If any experts or core devs could enlighten me on this matter (perhaps there's something I missed? perhaps there's a better way to do things?), that would be amazing.

Thanks!
leontristain

**Comment by leontristain** [comment]
Thanks for the quick masterlist fix!

I noticed that in the change, the syntax is:

```
    inc:
      - <<: *LotD_v5
        condition: 'not active("Solitude Skyway SE - Legacy of the Dragonborn Patch.esp")'
```

Now I'm wondering whether I can add the `condition:` to such a record from the userlist? Perhaps I could've done that locally if masterlist hasn't been updated?

It's like 1AM here so I will continue looking and experimenting tomorrow. I'd also love it if someone can comment on a way to ignore errors on `--auto-sort`. Again thanks!

**Comment by Ortham** [comment]
No, you can't use YAML merge keys (that's the `<<:` syntax) to reference an anchor (the `*LotD_v5`) from another document.

As for ignoring errors during `--auto-sort`, that's not possible by design. An error is supposed to occur when something is definitely wrong, and users should always be made aware of them. An intended effect of the current behaviour is that you've informed us that there is an issue with an incorrect error message, and CptMcSplody was able to fix that for everyone. If it was possible for the user to hide an error message that they'd already seen, as in #1083, I think it would be reasonable for `--auto-sort` to ignore hidden errors, but as LOOT can't tell if you've already seen an error or not, the safe thing is to fail if any are encountered.

**Comment by leontristain** [comment]
Hey! Thanks for the response.

If I'm understanding you right, the blocking issue is that LOOT does not wish to allow users (automation included) to ignore errors that a human has not already read. I suppose this is reasonable. It's a lot stricter than most tools I'm familiar with, but those aren't solving the problem you guys are solving so maybe there's a good case for it.

In that case, I'd like to suggest for the eventual design to be letting the user to specify the solution to the incompatibility as a userlist override, such that instead of ignoring errors, the user can choose to respond by actually addressing the problem whenever possible. There could even be a one-click button (maybe with a confirmation box to prevent accidental clicks) that says "submit this userlist override for masterlist consideration", that would automatically file a github issue, if the goal is to encourage feedback.

A userlist override would be nice, not only because it decouples automation getting unblocked from masterlist getting updated, but also because there are very legitimate cases where feedback should not happen but the actual check is still useful. For example, the user might have a personal patch that they either do not wish to share or simply does not make sense to share (such as an omni-patch that only makes sense with the user's specific load order). Such a patch would not be expected to be uploaded to the Nexus or other sites, and yet is nevertheless a valid solution to the incompatibility. In that scenario, it would be nice if the user can, in the userlist, say "this reported incompatibility in the masterlist is addressed by `MyCustomPatch.esp`, so please take `MyCustomPatch.esp` into account when checking for incompatibilities.

As it stands, the `MyCustomPatch.esp` scenario would be 100% an unaddressable blocker for automation under the current design. It would be nice if future design can find some way to allow this to be automated.

Anyway, these are just thoughts for your consideration. LOOT is a very useful tool that I'm very grateful to have. To be honest I was ecstatic when I saw that LOOT had `--auto-sort` so I did not have to reinvent the wheel for automating load order sorting. Thanks so much for your time and effort!

**Comment by Ortham** [comment]
I'm going to close this as `wontfix` as I stand by my decision to cancel auto-sort on errors. The points raised against this overlap with #1149 but I think provide a stronger motivation than the examples discussed in that issue, and with a narrower scope. I can imagine being able to override an `inc` entry by creating a user metadata entry for the same filename but a different condition that evaluates to false, though there are still a few issues with that. In any case, that's a change that's broader in scope than this issue and it isn't something I have any plans to investigate.
